composer/trainer/trainer.py

```diff
@@ -141,8 +141,13 @@ def _set_evaluator_interval_and_subset_num_batches(
             evaluator.eval_interval = eval_interval
 
 
-def _is_adaptive_grad_accum(grad_accum: Union[int, str], device: Device):
+def _is_adaptive_grad_accum(grad_accum: Union[int, str], device: Device, profiler: Optional[Profiler]):
```

Comment:
seems a bit odd that this method returns whether something is grad accum, but then has this weird side effect of checking the profiler. Can we just add this into the main trainer init, and accept the duplication?